webnotifications/webnotifications.d.ts

```diff
@@ -0,0 +1,116 @@
+﻿
+
+interface Notification {
+
+    /**
+     * The text displayed in the notification body
+     * @readonly
+     */
+    body?: string;
+
+
+    /**
+     * The direction the popup will show
+     * @readonly
+     */
+    dir?: "auto" | "ltr" | "rtl";
+
+    /**
+     * The icon shown in the notification
+     * @readonly
+     */
+    icon?: string;
+
+    /**
+     * The title shown in the notification
+     * @readonly
+     */
+    title?: string;
+
+    /**
+     * The notification language
+     * @readonly
```

Comment:
You can use the `readonly` keyword to specify a readonly property. `readonly lang?: string;`.